.github/workflows/build-and-test-nightly.yml

```diff
@@ -0,0 +1,23 @@
+name: Build and Test Nightly
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [master]
+  push:
+    branches: [master]
```

Comment:
because it's already here, you may change the event to be on push on `**`, and let's just remove the on pull_request event. It's safe to do this.
```yaml
on:
  push:
    branches: ['**']
```